### ISSUE TITLE
can't connect to api-server via service ip after useing kube-router with hostport and hairpin-mode

### ISSUE THREAD

**Issue by DalekFrog** [cloudnativelabs/kube-router#638]
First ,I used [this yaml](https://github.com/cloudnativelabs/kube-router/blob/master/daemonset/kubeadm-kuberouter-all-features.yaml) to deploy it. And everything works well。

when i found Ingress-nginx can‘t listen on  hostport (443/80) ,I replaced with another [yaml](https://github.com/cloudnativelabs/kube-router/blob/master/daemonset/kubeadm-kuberouter-all-features-hostport.yaml), and addnig `hairpin-mode=true`

After restarting kube-router ，Ingress-nginx and coredns couldn't connect to api-server 

some message
------------
version
 - OS: ubuntu 1604
 - kubeadm : 1.13.2
 - kubelet: 1.13.2
 - docker:  17.03.2
 - kube-router: latest

```sh
$ kubectl get svc  |grep kube
kubernetes                                 ClusterIP   100.80.0.1       <none>        443/TCP     36h


$ sudo ipvsadm -ln
IP Virtual Server version 1.2.1 (size=4096)
Prot LocalAddress:Port Scheduler Flags
  -> RemoteAddress:Port           Forward Weight ActiveConn InActConn
TCP  100.80.0.1:443 rr
  -> 10.40.10.53:6443             Masq    1      0          0
TCP  100.80.0.10:53 rr
  -> 100.64.0.3:53                Masq    1      0          0
TCP  100.80.204.124:5000 rr
  -> 100.64.1.34:80               Masq    1      0          0
TCP  100.84.11.212:80 rr
  -> 100.64.1.31:80               Masq    1      0          0
TCP  100.91.63.230:80 rr
  -> 100.64.1.2:80                Masq    1      0          0
```
```sh
$ curl 100.80.0.1:6443 -vI
* Rebuilt URL to: 100.80.0.1:6443/
*   Trying 100.80.0.1...
* Connected to 100.80.0.1 (100.80.0.1) port 6443 (#0)
> HEAD / HTTP/1.1
> Host: 100.80.0.1:6443
> User-Agent: curl/7.47.0
> Accept: */*
>

* Connection #0 to host 100.80.0.1 left intact


$ curl 100.80.0.1:443 -vI
* Rebuilt URL to: 100.80.0.1:443/
*   Trying 100.80.0.1...
* connect to 100.80.0.1 port 443 failed: Connection refused
* Failed to connect to 100.80.0.1 port 443: Connection refused
* Closing connection 0
curl: (7) Failed to connect to 100.80.0.1 port 443: Connection refused


$ curl 10.40.10.53:6443 -vI
* Rebuilt URL to: 10.40.10.53:6443/
*   Trying 10.40.10.53...
* Connected to 10.40.10.53 (10.40.10.53) port 6443 (#0)
> HEAD / HTTP/1.1
> Host: 10.40.10.53:6443
> User-Agent: curl/7.47.0
> Accept: */*
>

* Connection #0 to host 10.40.10.53 left intact
```

I am confusing about the proxy of service . Is there anyone can help me?
---

**Comment by murali-reddy** [comment]
Both hostport and hairpin mode functionality should not affect service proxy functionality. Are you able to easily reproduce it? I tried steps you mentioned I am not able to see any issue.

If you revert back to original yaml does issue go away?

**Comment by rmb938** [comment]
I ran into a similar issue but I assume it was related to this issue: https://github.com/cloudnativelabs/kube-router/issues/282. I have not tried this on the latest v0.2.5 release though

I can reproduce it with the following steps using v0.2.3 kube-router.

1. Spin up a K8s cluster using kubeadm and install kube-router v0.2.3
1. Launch a pod and query the kubernetes service.
    ```
    kubectl run -it --image=alpine sh
    apk -U add curl
    curl -k https://kubernetes
    {
      ...
      "code": 403
    }
    ```
1. Launch a pod that has a host port of 443 (i.e nginx ingress)
1. Launch a similar pod in step #2 on the same node as the port that has the host port.
    ```
    kubectl run -it --image=alpine sh
    apk -U add curl
    curl -k https://kubernetes
    - this curl just hangs
    ```

The only solution that I found is to just use a hostport that doesn't conflict with any service port.
